dev/manual_tests/lib/text.dart

```diff
@@ -572,7 +569,10 @@ class _UnderlinesState extends State<Underlines> {
                   vertical: size.height * 0.1,
                 ),
                 child: ListBody(
-                  children: lines,
+                  children: <Widget>[
+                    _wrap(null),
+                    for (TextDecorationStyle style in TextDecorationStyle.values) _wrap(style),
```

Comment:
I feel like we should decide if these inlined statements should always be formatted as one line like here or as two lines as seen elsewhere in the PR.

For me, the two liners make it easier to see at a glance that there's actually more magic happening in this collection.

Reply:
We also should decide whether it's acceptable for a `for` statement to have more than one line following it

Reply:
For styling `if`/`for` I tried to be consistant but with some rules:

* if `if`/`for` expression is _simple_: inlined
  ```dart
  List<String> l1 = <String>[
    'a',
    if (b) 'b',
  ];
  List<String> l2 = <String>[
    '1',
    for (String v in l1) v,
    '2',
  ];
  ```
* if there's _simple_ `if`/`else`: inlined but `else` on new line
  ```dart
  List<String> l = <String>[
    'a',
    if (b) 'b'
    else 'c',
  ];
  ```
* if then is multi-line: split
  ```dart
  List<String> l = <String>[
    'a',
    if (c)
      getString(
        long,
        params,
        list,
      ),
  ];
  ```
* if there are other splitted `for`/`if` in the list/map: split
  ```dart
  List<String> l = <String>[
    'a',
    if (b)
      'b',
    if (c)
      getString(
        long,
        params,
        list,
      ),
  ];
  ```
* if several `for`/`if` in the same expression: split
  ```dart
  List<String> l = <String>[
    for (String v in list)
      if (condition(v))
        'b',
  ];
  ```
* special cased multiline spreads
  ```dart
  List<String> l = <String>[
      if (condition(v)) ...<String>[
        'a',
        'b',
      ],
      for (String v in list) ...<String>[
        'c',
        v,
        'd',
      ],
  ];
  ```